File: ios/App/App/NewnodePlugin.m
//
//  NewnodePlugin.m
//  App
//
//  Created by Tony Rewin on 16.03.2022.
//

#import <Capacitor/Capacitor.h>

CAP_PLUGIN(NewnodePlugin, "Newnode",
    CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(start, CAPPluginReturnPromise);
)

File: ios/App/App/NewnodePlugin.swift
//
//  NewnodePlugin.swift
//  App
//
//  Created by Tony Rewin on 16.03.2022.
//

import Foundation
import Capacitor
import NewNode

@objc(NewnodePlugin)
public class NewnodePlugin: CAPPlugin {
    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve(["value": value])
    }
    @objc func start(_ call: CAPPluginCall) {
        let config = URLSessionConfiguration.default
        config.connectionProxyDictionary = NewNode.connectionProxyDictionary
        let session = URLSession(configuration: config)
        call.resolve(session) // TODO: check if it can return object
    }
}
